core/extension/gdextension_manager.h

```diff
@@ -63,6 +66,7 @@ class GDExtensionManager : public Object {
 
 public:
 	LoadStatus load_extension(const String &p_path);
+	LoadStatus load_function_extension(const String &p_path, GDExtensionConstPtr<const GDExtensionInitializationFunction> p_init_func);
```

Comment:
Another naming nitpick: "function extension" is an unclear term that we don't really use anywhere else.

Maybe `load_extension_from_function()`?

Or, if we can decide on an overall name for this feature we could use that term. So, if we standardize on "LibGodot", then this could be `load_libgodot_extension()`? Or, if we standardize on "embedded", then it could be `load_embedded_extension()`?